source/Halibut/Queue/QueuedDataStreams/HeartBeatMessage.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using Halibut.Queue.Redis.MessageStorage;
+using Newtonsoft.Json;
+
+namespace Halibut.Queue.QueuedDataStreams
+{
+    public class HeartBeatMessage
+    {
+        /// <summary>
+        /// Number of bytes of each DataStream has been uploaded to the service.
+        /// </summary>
+        public Dictionary<Guid, long> DataStreamProgress = new();   
```

Comment:
I'm still uneasy about this being a public mutable field, but as this is primarily used as a DTO, its impact is minimal.

Serialization shouldn't require this either.

Reply:
Changed to IReadOnlyDict